ui/src/state/friends.rs

```diff
@@ -21,3 +21,23 @@ pub struct Friends {
     #[serde(skip)]
     pub outgoing_requests: HashSet<Identity>,
 }
+
+impl Friends {
+    pub fn join(&mut self, other: &mut Friends) {
+        for (k, v) in other.all.drain() {
+            self.all.insert(k, v);
+        }
+
+        for v in other.blocked.drain() {
+            self.blocked.insert(v);
```

Comment:
ditto. Could have `other` be owned by this function and extend `self.block` (as well as others) with the fields in `other` if `other` isnt going to be reused.